File: software/examples/vbnet/ExampleInterrupt.vb
Imports System
Imports Tinkerforge

Module ExampleInterrupt
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your IO-4 Bricklet 2.0

    ' Callback subroutine for input value callback
    Sub InputValueCB(ByVal sender As BrickletIO4V2, ByVal channel As Byte, _
                     ByVal changed As Boolean, ByVal value As Boolean)
        Console.WriteLine("Channel: " + channel.ToString())
        Console.WriteLine("Changed: " + changed.ToString())
        Console.WriteLine("Value: " + value.ToString())
        Console.WriteLine("")
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim io As New BrickletIO4V2(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Register input value callback to subroutine InputValueCB
        AddHandler io.InputValueCallback, AddressOf InputValueCB

        ' Set period for input value (channel 1) callback to 0.5s (500ms)
        io.SetInputValueCallbackConfiguration(1, 500, False)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/go/example_interrupt.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/io4_v2_bricklet"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your IO-4 Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	io, _ := io4_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	io.RegisterInputValueCallback(func(channel uint8, changed bool, value bool) {
		fmt.Printf("Channel: %d\n", channel)
		fmt.Printf("Changed: %s\n", changed)
		fmt.Printf("Value: %s\n", value)
		fmt.Println()
	})

	// Set period for input value (channel 1) callback to 0.5s (500ms).
	io.SetInputValueCallbackConfiguration(1, 500, false)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
